ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterMetrics.java

```diff
@@ -83,6 +80,10 @@
   public final Histogram putManagerHandleResponseTimeMs;
   public final Histogram getManagerHandleResponseTimeMs;
   public final Histogram deleteManagerHandleResponseTimeMs;
+  // time spent in getting a chunk filled once it is available. This is reported per chunk.
+  public final Histogram chunkFillTimeMs;
```

Comment:
Considered both approaches and at the time I felt that calculating at the chunk level might skew the metrics due to some waits for chunks in an operation being much different than other waits. I now agree with you that it makes more sense to calculate this too at the chunk level. Will update.